docs/manual/key-storage/storage-plugins/azure-vault.md

```diff
@@ -0,0 +1,115 @@
+   - **Azure Client Secret**: The Azure Client Secret. 
+9. Choose `true` in **Show Allowed Only** if you want to hide keys based on their `Activation date` and `Expiration date` values. Disabled keys are always hidden.
+![Show Allowed Only](/assets/img/azure-vault-show-allowed.png)
+8. Click **Save** to commit changes for the integration.
+9. Click **Save** to add this integration to the Key Storage configuration.
+![Save config](/assets/img/azure-vault-save-config.png)
+
+Within the **Keys** tab, refresh the browser page and the new directory path specified in **Step 6** should appear:
+
+![Secrets Folder](/assets/img/azure-vault-secrets-folder.png)<br>
+
+Click into this directory and begin to navigate the secrets retrieved from Azure Key Vault.
+
+## Usage
+
+Once the integration is configured, the secrets from Azure Key Vault can be used by the various functions of the Runbook Automation product that require secrets.
+
+### Azure Key Vault Secret Naming Limitations
+
+Secret names in Azure Key Vault are restricted to alphanumeric characters and dashes.
+
+:::info Directories within Key Storage
+To enhance secret browsing in Azure Key Vault, secrets containing two consecutive dashes (`--`) will be organized into directories.
+
+For example, a secret named `kubernetes--clusters--api-token-12345` in Azure Vault will appear in Runbook Automation as:
+- Directory: `kubernetes`
+  - Subdirectory: `clusters`
+    - Key: `api-token-12345`
+
+When creating keys through the Rundeck Key Storage UI, you can use directories and the standard separator `/` normally. The system will automatically replace this with the allowed sequence `--`.
+:::
+
+### Rundeck Keys and Passwords
+
+Rundeck supports three key types: *Private Keys, Public Keys, and Passwords.* As private keys cannot be retrieved from Azure Key Vault Keys, all three kinds of Rundeck keys are stored as Azure Key Vault Secrets with the necessary metadata stored using Azure object Tags.
+
+Existing secrets in Azure Vault are treated as Rundeck Passwords by default, unless a specific tag is set:
+- `Rundeck-key-type: private` for private keys
+- `Rundeck-key-type: public` for public keys
+:::info File Extensions
+Due to Azure secret naming limitations, file extensions cannot be saved. For example, a public key named `my-key.pem` should be saved without the `.` and file extension in Rundeck.
+:::
+
+### Azure Key Vault Secret Versions
+
+While Azure supports versioning for Keys/Secrets, this plugin always retrieves the latest version. Updating an existing key through Rundeck generates a new version, ensuring the most recent version contains the updated value.
+
+:::info Updating Secrets
+Updating a secret using Rundeck Key Storage will only modify Rundeck-related tags.
+:::
+
+### Expired and Not-Yet-Enabled Secrets
+
+Azure Key Vault allows setting Activation and Expiration dates for keys and secrets. You can choose to respect these dates using the **Show Allowed Only** plugin configuration:
```

Comment:
this needs to be changed to match the new label https://github.com/rundeckpro/rundeckpro/pull/4111